### PR TITLE
test: Code Coverage: Add Edit Expense Page #5 Part 1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 2ebbd62</samp>

No summary available (Limit exceeded: required to process 192928 tokens, but only 50000 are allowed per call)

### <samp>🤖 Generated by Copilot at 2ebbd62</samp>

> _`AddEditExpensePage` is the gate of doom_
> _You must confirm your violation or face the gloom_
> _Category autofill is your only friend_
> _But don't forget to attach or it's the end_

### <samp>🤖 Generated by Copilot at 2ebbd62</samp>

No walkthrough available (Limit exceeded: required to process 192928 tokens, but only 50000 are allowed per call)

## Clickup
https://app.clickup.com/t/85zt9ph1c

## Code Coverage
`50.1% Statements 727/1451`
`44.32% Branches 687/1550`
`43.82% Functions 188/429`
`50.78% Lines 712/1402`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -3423,6 +3423,65 @@ export class AddEditExpensePage implements OnInit {
     });
   }
 
+  criticalPolicyViolationErrorHandler(err, customFields$) {
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3426]: Dont pass the observable, subscribe to it in the parent and pass the emitted value

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -3423,6 +3423,65 @@ export class AddEditExpensePage implements OnInit {
...
+            )
+          );
+        } else {
+          return throwError('unhandledError');
+        }
+      })
+    );
+  }
+
+  policyViolationErrorHandler(err, customFields$) {
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3445]: same comment here

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 3]
```diff
@@ -445,5 +455,254 @@ export function TestCases3(getTestBed) {
...
+            {
+              policyViolations: criticalPolicyViolation1,
+            },
+            of(customFieldData2)
+          )
+          .subscribe(() => {
+            expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+            expect(loaderService.showLoader).toHaveBeenCalledTimes(1);
+            expect(component.continueWithCriticalPolicyViolation).toHaveBeenCalledOnceWith(criticalPolicyViolation1);
+            expect(component.generateEtxnFromFg).toHaveBeenCalledTimes(1);
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:500]: Test case missing for throwerror

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 4]
```diff
@@ -445,5 +455,254 @@ export function TestCases3(getTestBed) {
...
+          )
+          .subscribe(() => {
+            expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+            expect(loaderService.showLoader).toHaveBeenCalledTimes(1);
+            expect(component.continueWithPolicyViolations).toHaveBeenCalledOnceWith(
+              criticalPolicyViolation1,
+              policyViolation1.data.final_desired_state
+            );
+            expect(component.generateEtxnFromFg).toHaveBeenCalledTimes(1);
+            done();
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:530]: same

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 5]
```diff
@@ -445,5 +455,254 @@ export function TestCases3(getTestBed) {
...
+          component: FyViewAttachmentComponent,
+          componentProps: {
+            attachments: [fileObjectData],
+            canEdit: true,
+          },
+          mode: 'ios',
+        });
+        expect(component.loadAttachments$.next).toHaveBeenCalledOnceWith();
+        expect(component.attachedReceiptsCount).toEqual(1);
+      }));
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:579]: edit case missing

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 6]
```diff
@@ -445,5 +455,254 @@ export function TestCases3(getTestBed) {
...
+        component.etxn$ = of(unflattenedExpData);
+        component.isConnected$ = of(true);
+        fileService.findByTransactionId.and.returnValue(of(fileObjectData[0]));
+        transactionOutboxService.fileUpload.and.resolveTo(fileObjectData);
+        fileService.post.and.returnValue(of(fileData1[0]));
+        spyOn(component, 'parseFile');
+        spyOn(component.loadAttachments$, 'next');
+        fixture.detectChanges();
+
+        component.attachReceipts({
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:701]: where are the expectations?

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3426]: The `generateEtxnFromFg ` requires an observable to be passed as custom fields, hence keeping it as it is.

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts (hunk 5 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:579]: This test case is for edit case. Add case is missing.

**Review by yksolanki9 — APPROVED** [review]
Test on device before merging

**Pull request merged**